modules/apps/31-ibc-query/types/msgs.go

```diff
@@ -10,25 +10,40 @@ const (
 )
 
 // NewMsgSubmitCrossChainQuery creates a new instance of NewMsgSubmitCrossChainQuery
-func NewMsgSubmitCrossChainQuery(path string, localTimeoutHeight, localTimeoutStamp, queryHeight uint64, clientId string) *MsgSubmitCrossChainQuery {
+func NewMsgSubmitCrossChainQuery(id string, path string, localTimeoutHeight uint64, localTimeoutStamp uint64, queryHeight uint64, clientId string, creator string) *MsgSubmitCrossChainQuery {
 	return &MsgSubmitCrossChainQuery{
+		Id:                 id,
 		Path:               path,
 		LocalTimeoutHeight: localTimeoutHeight,
 		LocalTimeoutStamp:  localTimeoutStamp,
 		QueryHeight:        queryHeight,
 		ClientId:           clientId,
+		Sender:             creator,
 	}
 }
 
+func (q MsgSubmitCrossChainQuery) GetQueryId() string { return q.Id }
+
+func (q MsgSubmitCrossChainQuery) GetPath() string { return q.Path }
+
+func (q MsgSubmitCrossChainQuery) GetTimeoutHeight() uint64 { return q.LocalTimeoutHeight }
+
+func (q MsgSubmitCrossChainQuery) GetTimeoutTimestamp() uint64 { return q.LocalTimeoutStamp }
+
+func (q MsgSubmitCrossChainQuery) GetQueryHeight() uint64 { return q.QueryHeight }
+
+func (q MsgSubmitCrossChainQuery) GetClientID() string { return q.ClientId }
+
+
 // ValidateBasic implements sdk.Msg and performs basic stateless validation
-func (msg MsgSubmitCrossChainQuery) ValidateBasic() error {
+func (q MsgSubmitCrossChainQuery) ValidateBasic() error {
 
 	return nil
 }
 
```

Comment:
It's not important but I was just curious.. why do you change msg to q?
https://docs.cosmos.network/v0.46/building-modules/messages-and-queries.html

sample code in doc uses msg